packages/protocol/lib/compatibility/version.ts

```diff
@@ -105,3 +124,57 @@ export class ContractVersionDelta {
     return deltas.map(d => d.toString()).join('.')
   }
 }
+
+/**
+ * A mapping {contract name => {@link ContractVersionDelta}}.
+ */
+export interface ContractVersionDeltaIndex {
+  [contract: string]: ContractVersionDelta
+}
+
+/**
+ * A mapping {contract name => {@link ContractVersion}}.
+ */
+export interface ContractVersionIndex {
+  [contract: string]: ContractVersion;
+}
+
+/**
+ * A mapping {contract name => {@link ContractVersion}}.
+ */
+export class ContractVersions {
```

Comment:
Regarding the file usage, we could have `version.ts` with all independent-from-openzeppeling types (contract version, delta, version checker), and then have `ast-version.ts` with things that use both version and artifacts stuff. If you agree, then move this type to `ast-version.ts` so we can remove the `ast-code.ts` dependency.

Reply:
Great suggestion!